grid/chat/get-messages/maps/discord.suma

```diff
@@ -0,0 +1,203 @@
+profile = "chat/get-messages@1.0"
+provider = "discord"
+
+"""
+GetMessages map
+"""
+map GetMessages {
+	destination = input.destination
+	afterId = undefined
+	beforeId = undefined
+
+	set if(input.afterDate) {
+		afterId = call GetSnowflake(date = input.afterDate)
+	}
+
+	set if(input.beforeDate) {
+		beforeId = call GetSnowflake(date = input.beforeDate)
+	}
+
+	http GET "/channels/{destination}/messages" {
+		request "application/json" {
+			query {
+				after = afterId
+				before = beforeId
+				limit = input.limit
+			}
+
+			headers {
+				"Authorization" = `Bot ${parameters.accessToken}`
+			}
+		}
+
+		response 200 "application/json" {
+			messages = call MapMessages(messages = body)
+			rateLimit = call MapRateLimitDetails(headers = headers)
+
+			map result {
+				previousPage = messages[0].id
+				nextPage = messages[messages.length - 1].id
+				messages = messages
+				rateLimit = rateLimit
+			}
+		}
+
+		response 400 "application/json" {
+			map error {
+				title = "Bad Request"
+				detail = body
+			}
+		}
+
+		response 401 "application/json" {
+			map error {
+				title = "Unauthorized"
+				detail = body.message
+			}
+		}
+
+		response 403 "application/json" {
+			map error {
+				title = "Forbidden"
+				detail = body.message
+			}
+		}
+
+		response 404 "application/json" {
+			map error {
+				title = "Not Found"
+				detail = body.message
+			}
+		}
+
+		response 405 "application/json" {
+			map error {
+				title = "Method not allowed"
+				detail = body.message
+			}
+		}
+
+		response 429 "application/json" {
+			map error {
+				title = "Too many requests"
+				detail = body.message
+			}
+		}
+
+		response 502 "application/json" {
+			map error {
+				title = "Gateway unavailable"
+				detail = body.message
+			}
+		}
+	}
+}
+
+"""
+Operation MapMessages
+Returns array mapping messages in received body
+"""
+operation MapMessages {
+	messages = args.messages.map(message => {
+		const result = {
+			id: message.id,
+			author: {
+				id: message.author.id,
+				username: message.author.username
+			},
+			createdAt: Date.parse(message.timestamp),
+			text: message.content
+		}
+		const discordEpoch = 1420070400000;
+		const getTimestamp = (snowflake) => {
```

Comment:
There is an operation GetTimestamp defined, is not this duplicity?

Reply:
I'll refactor it